### ISSUE TITLE
ZeroMq doesn't build for Electron 9

### ISSUE THREAD

**Issue by aminya** [n-riesco/jmp#31]
https://github.com/nteract/hydrogen/issues/2034

Edit:

I pushed a fix to ZeroMq itself. See this:
https://github.com/zeromq/zeromq.js/pull/444

**Comment by n-riesco** [comment]
I can't tell from https://github.com/nteract/hydrogen/issues/2034 . Is your visual studio installation able to build v140 projects?

**Comment by aminya** [comment]
@n-riesco I pushed a fix to ZeroMq itself. See this:
https://github.com/zeromq/zeromq.js/pull/444

**Comment by n-riesco** [comment]
Note, though, that `zeromq@6` has been in beta release for a long time, and that `jmp` still depends on `zeromq@5`.

**Comment by maclockard** [comment]
Unfortunately it looks like active development on zeromq.js is at least temporarily on pause:
 * https://github.com/zeromq/zeromq.js/issues/436
 * https://github.com/zeromq/zeromq.js/issues/442
 * https://github.com/0rpc/zerorpc-node/pull/111

**Comment by firminochangani** [comment]
Hi @maclockard,

Is there any reason why is this happening?

Thanks.

**Comment by maclockard** [comment]
@flowck I'm not associated with zermomq, so I do not know why this is the case, sorry.

**Comment by aminya** [comment]
@n-riesco Should I register my fork, and make a PR here? It might take some time until they review the PR.

**Comment by n-riesco** [comment]
@aminya I wouldn't rush into forking. Let's try to figure out who can make a release (this is not the first time we've had this kind of problem).

I've checked https://www.npmjs.com/package/zeromq and @captainsafia is listed as a collaborator (doesn't that mean she could do a release?).

In the meantime, so that we don't hold your progress, instead of forking, we could [use a github URL in package.json](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#github-urls). I could create a branch pointing to your branch.

How do you think? Would that work for you?

---

PS: I'm sorry I haven't reviewed your PR for hydrogen (I was planning to do it yesterday, but something came up).

**Comment by aminya** [comment]
Thank you. The code runs `prepare` and `build` scripts before publishing. We cannot use a GitHub link in NPM as the repository itself doesn't include the built files.

Atom 1.56 was released today with Electron 9, which means we are already late, and we need to update Hydrogen as soon as possible.

P.SL I also see @lgeiger in the commit list, which may help us fix the issue.

**Comment by aminya** [comment]
I also made a PR for Zeromq v5. It seems to fix the build issue. 
https://github.com/zeromq/zeromq.js/pull/445

**Comment by captainsafia** [comment]
> I've checked https://www.npmjs.com/package/zeromq and @captainsafia is listed as a collaborator (doesn't that mean she could do a release?).

Yes, I have access to the zeromq.js package and have maintainer permissions on the repo since I was contributing during its inception.

I haven't personally been contributing to it for several years but it looks like we need to find someone to help out with this.

Assuming that we can validate #445, I'm happy to review, merge, and ship out a new release for this.

**Comment by n-riesco** [comment]
The PR for v5 would be brilliant by me (as I wouldn't need to change anything).

For v6, I've created a branch, `zeromq6` and registered the package `jmp-zeromq6`.
I'd be happy to point `jmp-zeromq6` to your fork if you want to test it.

**Comment by n-riesco** [comment]
`zeromq@6.0.0-beta.6` fails to build on Travis: https://travis-ci.org/github/n-riesco/jmp/jobs/767029456

```
-- Installing: /home/travis/build/n-riesco/jmp/node_modules/zeromq/build/libzmq/share/cmake/ZeroMQ/ZeroMQConfigVersion.cmake

make[1]: Leaving directory '/home/travis/build/n-riesco/jmp/node_modules/zeromq/build/libzmq'

  TOUCH Release/obj.target/libzmq.stamp

  CXX(target) Release/obj.target/zeromq/src/context.o

In file included from ../src/module.h:8:0,

                 from ../src/context.cc:3:

../src/util/arguments.h:6:20: fatal error: optional: No such file or directory

compilation terminated.

zeromq.target.mk:123: recipe for target 'Release/obj.target/zeromq/src/context.o' failed

make: *** [Release/obj.target/zeromq/src/context.o] Error 1

make: Leaving directory '/home/travis/build/n-riesco/jmp/node_modules/zeromq/build'
```

**Comment by n-riesco** [comment]
Fixed: zeromq fails to build in xenial. I've set Travis to use bionic.

**Comment by n-riesco** [comment]
Fixed in [`zeromq@5.2.6`](https://github.com/zeromq/zeromq.js/releases/tag/v5.2.6).
